### PR TITLE
Docsteam 31 execution env pages preview

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
* I have rearranged our execution environments docs to mirror the way the options are shown on the website: https://circleci.com/execution-environments/ ... Docker, Linux, macOS, Windows, GPU, Arm
* I've made separate pages for each execution environment.
* Tried to clear up language - previously we have not been explicit about the difference between execution environment and executor. Both Linux, Windows, GPU, Arm etc all use the `machine` executor, for example.
* I have archived the "Choosing your Executor type" doc because there was so much overlap between that and the "Introduction to Execution Environments" - I've split up the content from the archived executor-types doc to create the new pages for each individual execution environment page.
* Fixed lots of links that were going to the archived page. Any that need a link the the `machine` executor specifically now go to the entry for that config key in the configuration reference. Links that need to go to an overview of different executors now go to the "Introduction to execution environments" doc

I have used a preview but it appear to be broken, probably due to the detangling project: http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/DOCSTEAM-31-execution-env-pages-preview/?force-all

# Reasons
Executors docs were confusing due to the difference between "execution environment" and "executor". One long doc was being used as the overview for multiple sections making navigation confusing.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by rsylvian** [comment]
> I have used a preview but it appear to be broken, probably due to the detangling project: http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/DOCSTEAM-31-execution-env-pages-preview/?force-all

@rosieyohannan This is still two different projects for now so this should not be related. Just a FYI :)

**Comment by rsylvian** [comment]
@rosieyohannan Like we discussed last time (for `server4`), I think we should hold off on creating subdirectories in the content until we are entirely on the new platform. My main reasons are:

- We are still waiting on SEO recommendations on how to structure URLs (they are against directories in the URL)
- Flattening URL is much harder to handle with jekyll that it is with our new platform
- It will allow Emma to be able to prioritize that work based on everything else we are trying to accomplish in Q2

Thoughts?

**Comment by rosieyohannan** [comment]
@rsylvian Yes agreed - I'll remove the subdirectory. I only used it here because I was creating entirely new pages but I'll remove for now. I did meet with marketing and they seemed fine with one dir deep URL. So having /server/ in there should be OK, and maybe some directories for other sections of the sidebar etc, just not multiple subdirectories deep. Just very keen to get some file structure asap 😅 . Thanks for your feedback!

File: jekyll/_cci2/executor-intro.md [hunk 1]
```diff
@@ -144,16 +167,121 @@ jobs:
       - run: Write-Host 'Hello, Windows'
 ```
 
-Find out more about the `windows` executor in the [Using the Windows executor]({{ site.baseurl }}/2.0/executor-types/#using-the-windows-executor) section of the Choosing an Executor Type page. See [the Windows orb details](https://circleci.com/developer/orbs/orb/circleci/windows) for the list of options available in the Windows orb.
+Find out more about the Windows execution environment in the [Using the Windows Execution Environment]({{ site.baseurl }}/2.0/execution-environments/using-windows) page. See [the Windows orb page in the developer hub](https://circleci.com/developer/orbs/orb/circleci/windows) for the list of options available in the Windows orb.
+
+## GPU
```
Comment (rosieyohannan) [inline jekyll/_cci2/executor-intro.md:172]:
```suggestion
## GPU
{: #gpu }
```

**Review by kbatuigas — APPROVED** [review]

**Pull request merged**